text/0000-long-error-codes-explanation-normalization.md

```diff
@@ -0,0 +1,143 @@
+
+Start Date: 2016-01-04
+
+RFC PR:
+
+Rust Issue: N/A
+
+# Summary
+
+Long error codes explanations haven't been normalized yet. For example, actually, not all error codes explanation provide an erroneous code example or a way to fix the error. This RFC intends to clarify this by providing a template to follow.
+
+# Motivation
+
+Long error codes explanations are a very important part of Rust. Having an explanation of what failed helps to understand the error and is appreciated by Rust developers of all skill levels. Providing an unified template is needed in order to help people who would want to write ones as well as people who read them.
+
+# Detailed design
+
+Here is what I propose:
+
+## Error description
+
+Giving a little more detailed error message. For example, with the following code:
+
+```Rust
+extern crate a;
+extern crate b as a;
+```
+
+We get the `E0259` error code which says "an extern crate named `a` has already been imported in this module" and the error explanation says: "The name chosen for an external crate conflicts with another external crate that has been imported into the current module.".
+
+## Minimal example
+
+Giving an erroneous code example which directly follows `Error description`. It'll be helpful for the `How to fix the problem`. Making it as simple as possible is really important in order to help readers to understand what the error is about. A comment should be added with the error on the same line that the errors happen. Example:
+
+```Rust
+type X = u32<i32>; // error: type parameters are not allowed on this type
+```
+
+ If the error comments is too long to fit 80 columns, split it up like this, so the next line start at the same column of the previous line:
+
+```Rust
+type X = u32<'static>; // error: lifetime parameters are not allowed on
+                       //        this type
+```
+
+And if the code line is just too long to make a correct comment, put your comment before it:
```

Comment:
"If the sample code is too long to write an effective comment, place your comment on the line before the sample code."